octokit/pagination.py

```diff
@@ -0,0 +1,38 @@
+from .resources import Resource
+
+class Pagination(object):
+  def __init__(self, *args, **kwargs):
+    # TODO (howei): possibly extract auto_paginate from kwargs
+    # so users can do client = Client(auto_paginate=True)
+    self.auto_paginate = False
+    self.last_response = None
```

Comment:
Yep because of how we set up mixins, each of the mixin's response_callbacks will be called in the order our mixins are declared:

`class Client(RateLimit, Pagination, ..)` - RateLimit's response callback will be called, then Pagination's then so forth and this will happen with every request because of the hook in session.

Since we can do that I thought it'd make more sense for RateLimit to initialize and and update last_response (as opposed to Client doing this).